gtdynamics/universal_robot/Robot.h

```diff
@@ -112,6 +112,32 @@ class Robot {
   /// Print links and joints of the robot, for debug purposes
   void print(const std::string &s = "") const;
 
+  /// Overload equality operator.
+  bool operator==(const Robot &other) const {
+    // Define comparators for easy std::map equality checking
+    auto link_comparator = [](decltype(*this->name_to_link_.begin()) a,
+                              decltype(a) b) {
+      // compare the key name and the underlying shared_ptr object
```

Comment:
nit: can you \_emph_ or make a note in the comment "Define comparators ..." or something that clarifies that the _only_ reason this is needed is for the `shared_ptr` ?  Spent a couple minutes trying to figure out why you didn't just use the default map comparator 😂

Reply:
I am still not a fan of this shared pointer implementation of Links and Joints. Other than efficiency (which can also be achieved with references), was there any other reason for choosing to use them?

Reply:
references won't work for polymorphic types, e.g. RevoluteJoint vs ScrewJoint vs PrismaticJoint

specifically, to hold a container of polymorphic types you need pointers (unless use fancy std::variant stuff but I don't think that suits our use case - inherently we need runtime polymorphism to hold all the joints in one data structure)

Reply:
Huh, I figured we could just pass references to the Joint class since the public API for each joint type is the same. I am not going to tackle this anytime soon so we can actually drop it. :sweat_smile: